### ISSUE TITLE
someone had tryed ELS27 with Megane 4?

### ISSUE THREAD

**Issue by igno77** [cedricp/ddt4all#76]
Hi to all,
I've seen in last version of ddt4all the new icon of ELS27. After a quick reserch of information I know that this interface is better than ELM (faster, stable and use Mcan and Lcan at the same time) but....we need to modify some wiring for use it with a Megane 4? It seems this interface born for Ford and Mazda cars, not for Renault. On a russian website there is a modification of the els27 but no one have had success with the connection. Do you have more info? What kind of chip we must have insiede?
Thanks

**Comment by KarelSvo** [comment]
You only must to modify the ELS27 if you want to use the media can bus. 
https://a.d-cd.net/40c326ds-960.jpg
Simply get yourself an experience with els27 from aliexpress from 13 euro.
Please your note, obdlink interface and els are preset to baudrate 115200. If firmware is not up-to-date then just update to scantool.net.
I must to use windows 10 for update  ELS.

**Comment by pakoto** [comment]
Check photo. Description says 3-12 11-13: wrong, and photo shows 3-13, 12-14: wrong because you connect canbus and mm canbus into the car obd, dangerous!).

**Comment by premultiply** [comment]
No. "Normal" CAN-Bus is on 6+14 (unmodified).
Only the second independant CAN-Interface needs to be briged to Renault MM-CAN on 12+13. Ford-Pins 3+11 does not have any connection on Renault vehicle OBD port so there is no need to remove or disconnect them on ELS27. Just wire these two connectors and you can use it on Ford and Renault vehicles.

**Comment by pakoto** [comment]
Sorry, i mean that zmiterm bridge 6-13 and 12-14 and use out the car. See green red cables in the photo and compare position with generic OBD pinout photo.

**Comment by ohayyam** [comment]
I tried many combinations but no success. Even i disconnect obd socket side and main board connections and made many more combinations according to ford and renault pin out diagrams but all fail. I never read succesfull story. does really anyone use els27 with any renault ? I already asked to zmiterm on drive2.ru no answer. I believe els27 may fake ??? I contact russian seller of original els27 he said 75 usd but he did not know about cable modification for renault. He said original els27 will work with renault. I am not sure and waiting to read success story with els27 and renault.

**Comment by ohayyam** [comment]
I already mail latest ddt4all v.3.4.5 program with mar2018 database to cedric to check ;) download link on drive2.ru site by zmiterm. 

My els27 may damage during combination try :)) never connected or it was fake because i tried it without cable mod. It did not work for me.

**Comment by premultiply** [comment]
http://ffclub.ru/topic/311647/

**Comment by premultiply** [comment]
@cedricp This looks like the protocol description to change the CAN interface for MM-CAN access:
https://www.scantool.net/downloads/118/mx_protocol_commands.pdf
You have to select a "MS-CAN" preset first an then adjust the baud rate to 500000.

**Comment by cedricp** [comment]
I saw that, but I don't have the hardware to test it yet.

**Comment by premultiply** [comment]
I have one here. How can I assist you?

**Comment by premultiply** [comment]
First try run (without connected to car X10 ZOE) with els27 branch:
```
Traceback (most recent call last):
  File ".\ddt4all.py", line 514, in changecanspeed
    self.paramview.setCanLine(item)
  File "C:\Program Files (x86)\ddt4all\parameters.py", line 534, in setCanLine
    self.initELM()
  File "C:\Program Files (x86)\ddt4all\parameters.py", line 537, in initELM
    connection_status = self.ecurequestsparser.connect_to_hardware(self.canline)
  File "C:\Program Files (x86)\ddt4all\ecu.py", line 1015, in connect_to_hardware
    options.elm.set_can_addr(short_addr, ecu_conf, canline)
  File "C:\Program Files (x86)\ddt4all\elm.py", line 1170, in set_can_addr
    self.lf.write('#' * 60 + "\n#connect to: " + ecu['ecuname'] + " Addr:" + addr + "\n" + '#' * 60
+ "\n")
TypeError: cannot concatenate 'str' and 'NoneType' objects
Scanning ECU FRRAD
Scanning ECU INV-ME_29b
Scanning ECU Aide au parking
Warning, address DB is not mapped
Warning, address DC is not mapped
Skipping CAN extended address (not supported yet)  DA
Scanning ECU WCGS
Scanning ECU FCAM
Scanning ECU BCB
Scanning ECU PEB
Scanning ECU CAN Adapter / BAC
Scanning ECU VSP
Scanning ECU AAU
Scanning ECU IDM
Scanning ECU UCH
Scanning ECU ABS-VDC - ABS-ESP
Scanning ECU Direction assiste
Scanning ECU HVAC
Scanning ECU LBC2
Scanning ECU HFM
Scanning ECU E-ACT-EBA_29b
Scanning ECU BCB-OBC_29b
Scanning ECU DCM Renault
Scanning ECU UBP
Scanning ECU UPC-EMM
Scanning ECU Frein de parking lectrique
Scanning ECU Airbag-SRS
Scanning ECU LBC (EV)
Scanning ECU (H) EVC
Scanning ECU PLC/PLGW
Skipping CAN extended address (not supported yet)  2D
Scanning ECU Audio
Scanning ECU MIU
Scanning ECU Navigation-UCC-ITM
Scanning ECU TDB
Skipping CAN extended address (not supported yet)  D2
```

**Comment by cedricp** [comment]
This error is not related to **els27** work, which ECU are you trying to connect ?

**Comment by cedricp** [comment]
Just commited a fix for this kind of problem.

**Comment by premultiply** [comment]
Ok, did a test on my car (ZOE) now but switching to "MediumSpeed CAN" does not change anything.
All communcation stays on HS-CAN.

Switched to "MediumSpeed CAN" after startup, selected my vehicle from list and pressed scan.
All ECUs from HS-CAN where found but nothing from MM-CAN (Navigation/MFD, ...) was found.

```
Verwendung einer benutzerdefinierten Datenbank
libpng warning: iCCP: known incorrect sRGB profile
Starten des ELM mit einer Baudrate von 500000...
Versuche Anschluss zu oeffnenCOM17 @ 500000
Checking els
Test der Baudrate: 500000
Starten mit Baudrate: 500000
2933 ECUs wurden geladen.
Scanning ECU FRRAD
Scanning ECU INV-ME_29b
Scanning ECU Aide au parking
Warning, address DB is not mapped
Warning, address DC is not mapped
Skipping CAN extended address (not supported yet)  DA
Scanning ECU WCGS
Scanning ECU FCAM
Scanning ECU BCB
Scanning ECU PEB
Scanning ECU CAN Adapter / BAC
Scanning ECU VSP
Scanning ECU AAU
Scanning ECU IDM
Scanning ECU UCH
Scanning ECU ABS-VDC - ABS-ESP
Scanning ECU Direction assiste
Scanning ECU HVAC
Scanning ECU LBC2
Scanning ECU HFM
Scanning ECU E-ACT-EBA_29b
Scanning ECU BCB-OBC_29b
Scanning ECU DCM Renault
Scanning ECU UBP
Scanning ECU UPC-EMM
Scanning ECU Frein de parking lectrique
Scanning ECU Airbag-SRS
Scanning ECU LBC (EV)
Scanning ECU (H) EVC
Scanning ECU PLC/PLGW
Skipping CAN extended address (not supported yet)  2D
Scanning ECU Audio
Scanning ECU MIU
Scanning ECU Navigation-UCC-ITM
Scanning ECU TDB
Skipping CAN extended address (not supported yet)  D2
```

**Comment by cedricp** [comment]
Did you rewire the MS-CAN lines on the OBD connector ? AFAIK, it's not connected to the good pins.

**Comment by cedricp** [comment]
Just fixed an issue with MS-CAN scan, sorry, can you retry ?

**Comment by premultiply** [comment]
Yes of course it's rewired.
Yep, I will try again in a few minutes.

**Comment by premultiply** [comment]
Sorry, still only ECUs from HS-CAN.
And manual MFD selection still does not get any data.

**Comment by premultiply** [comment]
There are some more command descriptions in https://www.scantool.net/scantool/downloads/234/stn1100-frpm-preliminary.pdf which are removed from later versions of the documentation.

**Comment by cedricp** [comment]
That's weird, can you send me your _elm log_ over ?

**Comment by cedricp** [comment]
You can also try to add a line at the end :
**elm.py 1192** 
```
        if canline == 0:
            if 'brp' in ecu.keys() and ecu['brp'] == "1":  # I suppose that brp=1 means 250kBps CAN
                self.cmd("AT SP 8")
            else:
                self.cmd("AT SP 6")
        else:
            self.cmd("STP 53")
            self.cmd("STPBR 250000") # this line
```

**Comment by premultiply** [comment]
No problem. Sent you the log by mail.
But there is no STP command inside at all :-(

**Comment by premultiply** [comment]
Mmmmh. I downloaded the changed files only. Maybe I've forgotten something...
I will try with full els27.zip again.

**Comment by premultiply** [comment]
Dry-run els27.zip without car connected still does show any STP commands in log.

**Comment by cedricp** [comment]
That is really strange... I checked the code several times...

**Comment by cedricp** [comment]
Ok, just found and fixed the problem... You can retry if you have time

**Comment by premultiply** [comment]
After several hours of testing I think all ST commands are sent out as raw data because they are filtered
https://github.com/cedricp/ddt4all/blob/7b28b65f613559ed1780ed2eb60af8d0cb7f244b/elm.py#L737 so the never reach the ELS27. This is why they never show up in any log...

**Comment by cedricp** [comment]
Well spotted ! I fixed that issue

**Comment by pakoto** [comment]
works multimedia canbus using bridges or still pending tests?

**Comment by cedricp** [comment]
Still waiting for last commit testing ... (_els27_ branch)

**Comment by premultiply** [comment]
Software side seems to be ok now.
But using the modified els27 on the car disturbed and crashed my radio ECU on MM-CAN. Had to pull the fuse to make it work again.

I am not able to verify that the modified hardware does what it should do.

**Comment by pakoto** [comment]
ford switch canhigh 6-3 and canlow 14-11: 3 and 11 are mm canbus.
renault switch CH 6-13 and CL 14-12: 13 and 12 are mm canbus.

For conversion i think you need to connect mm CH 3-13 and mm CL 11-12. Other combination connect mm canhigh with mm canlow, or worse shortcircuit canbus with mm canbus.

IMHO

**Comment by premultiply** [comment]
Yes it seems you are right. The polarity is wrong.
I have to change it on my ELS27.

**Comment by pakoto** [comment]
Remember that obd pinout front view images from internet are mirrored with els27 front view.

I check with multimeter pins using this ELS27 front view 

\   .. 8  ..  7 .   6  .  5  .  4  .  **3** . 2 .  1  .  /
.  \ 16 15 **14 13** 12 .**11**. 10  .9 /

**Comment by premultiply** [comment]
The pin numbers are always the same for rear or front view  ;-)

**Comment by pakoto** [comment]
@premultiply i need your help with off-topic  if you are creator of ZOEdisplay:
We are trying to add DPF soot level into dacia-renault DCIs using arduino. ddt2000 database obtain it using:

Send 22242C
Receive 62242C0000
Start byte: 4
Bits: 16
Unit: gr
Value: (AX+B)/C
A=1
B=0
C=100

Do you know how to translate to arduino code using mcp2515 module? need to ask every x secs to ecu?

**Comment by premultiply** [comment]
@pakoto I don't know how to poll data from ECUs. Never thought about that. All I need to do for ZOEdisplay is listen to the bus.

**Comment by pakoto** [comment]
i don´t know how it works. When we connect to motor ecu using ddt4all, we can get too much sensor info than using stardard tools like android torque. We are trying to take some parameters and send to arduino lcd.

**Comment by cedricp** [comment]
It's because each ECM (engine control module) CAN ISO_TP frames are differents. This is why you need an engineer tool like DDT4all to get them.
Select an ECU and look at _request_ and _data_ tabs in **edition** mode to know how frames are built.

**Comment by cedricp** [comment]
I started a project for Arduino UNO with MCP2515 here : 
https://gitlab.com/DrQuark/x84_can
There are ISO_TP and CAN_RAW sources examples.

**Comment by cedricp** [comment]
Send 22242C
Receive 62242C0000
Start byte: 4
Bits: 16
Unit: gr

as the wanted value starts at byte 4 (index 3) and we have 16 bytes (2 bytes) :

receiv_buffer = {0x62, 0x24, 0x2C, 00, 00};
short my_value = (receiv_buffer[3] << 8) | receiv_buffer[4]);

should be good

**Comment by premultiply** [comment]
New modification 3->13 and 11->12. Only one wire needed ;-)
![img_3241](https://user-images.githubusercontent.com/4681172/41549336-d3cb3058-7325-11e8-9dbf-62051354a9cb.jpg)
![img_3242](https://user-images.githubusercontent.com/4681172/41549337-d3e61b5c-7325-11e8-96c9-5cb8e2bb28cd.jpg)
![img_3243](https://user-images.githubusercontent.com/4681172/41549338-d40441ea-7325-11e8-95d6-41cf27094fed.jpg)

**Comment by pakoto** [comment]
Yes. I have same mod. Works?

**Comment by premultiply** [comment]
Yes, hardware does work now!
But the ddt4all part is not working correctly.

First problem was: MM-CAN on my ZOE car is 500 kbd not 250 kbd. I fixed it manually in code ("ST PBR 500000")
2nd unsolved problem: But even with that modification only parts of the ECU data is retrieved by ELS27 compared to an modified ELM327 on MM-CAN. But my ELS27 STN117 has FW 3.3.1 and I am not able to update it to lastest FW 4.2.4 (https://www.scantool.net/downloads/updates/stn1170/).

**Comment by pakoto** [comment]
i can´t update my fw either. Aliexpress ELS27 clone i suppose.

I´ll test with my dacia dokker connecting to medianav and post results.

**Comment by pakoto** [comment]
minor issue: elm benchmark don´t work with this branch.

**Comment by pakoto** [comment]
Test elm fails
Connect works

remember, minor issue. els27 don´t need elm check because it works.

![capture1](https://user-images.githubusercontent.com/24203270/41583272-34fbb450-73a4-11e8-8540-6be302e5690a.JPG)
![capture2](https://user-images.githubusercontent.com/24203270/41583273-351a831c-73a4-11e8-98b3-f99fe8c7cf87.JPG)

**Comment by premultiply** [comment]
And by the way the serial port speed selection dropdown for ELS27 is useless. ddt4all does only auto detection. But the autodetection should run bitrates 2000000, 1000000, 500000, 115200, 38400.

The ELM test seems to try a wrong port etc. Same result for me.

**Comment by cedricp** [comment]
Ok, I see

**Comment by pakoto** [comment]
yes, port selection could get some improvements:
-bluetooth, usb, and obdlink selection could be one. Works selecting COMx and usb icon. Tested.
-option to remember expert mode and check recommendations. You could add a big scary message first time and a "don´t show again" checkbox.

**Comment by cedricp** [comment]
```
yes, port selection could get some improvements:
-bluetooth, usb, and obdlink selection could be one. Works selecting COMx and usb icon. Tested.
```
Ok
```
-option to remember expert mode and check recommendations. You could add a big scary message first time and a "don´t show again" checkbox.
```
Not sure it's a good idea, I already had bad surprises when working on the first version when expert mode was not yet in place...

**Comment by premultiply** [comment]
Here is some log extract from ELS27 session tests to MFD on ZOEs MM-CAN. You can see that some commands do not get an answer. Maybe the specific AT commands needs to be sent AFTER MS-CAN preset call?

```
>[19:19:51.504]ATZ
<[0.874]ATZ


ELM327 v1.3a

>
############################################################
#[06/18/18 19:20:24.632] Init CAN
############################################################
>[19:20:24.632]AT WS
<[0.031]AT WS


ELM327 v1.3a

>
>[19:20:24.663]AT E1
<[0.015]AT E1
OK

>
>[19:20:24.678]AT S0
<[0.016]AT S0
OK

>
>[19:20:24.694]AT H0
<[0.016]AT H0
OK

>
>[19:20:24.710]AT L0
<[0.015]AT L0
OK

>
>[19:20:24.725]AT AL
<[0.016]AT AL
OK

>
>[19:20:24.741]AT CAF0
<[0.015]AT CAF0
OK

>
############################################################
#connect to: MFD v4.9 Addr:58
############################################################
>[19:20:24.756]AT SH 747
<[0.016]AT SH 747
OK

>
>[19:20:24.772]AT CRA 767
<[0.016]AT CRA 767
OK

>
>[19:20:24.788]AT FC SH 747
<[0.015]AT FC SH 747
OK

>
>[19:20:24.803]AT FC SD 30 00 00
<[0.016]AT FC SD 30 00 00
OK

>
>[19:20:24.819]AT FC SM 1
<[0.015]AT FC SM 1
OK

>
>[19:20:24.834]STP 53
<[0.016]STP 53
OK

>
>[19:20:24.850]STPBR 500000
<[0.016]STPBR 500000
OK

>
>[19:20:24.881]0210C0
<[0.141]0210C0
0250C0FFFFFFFFFF

>
>[19:20:26.020]023E01
<[0.125]023E01
017EFFFFFFFFFFFF

>
>[19:20:27.533]023E01
<[0.125]023E01
017EFFFFFFFFFFFF

>
>[19:20:29.046]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:20:30.560]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:20:32.073]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:20:33.586]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:35.099]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:20:36.612]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:20:38.126]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:20:39.639]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:41.152]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:42.665]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:44.179]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:20:45.692]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:20:47.205]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:48.718]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:50.231]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:20:51.745]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:20:53.258]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:20:54.771]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:56.284]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:57.797]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:20:59.311]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:21:00.824]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:02.337]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:03.850]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:05.363]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:21:06.877]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:21:08.390]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:09.903]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:11.416]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:12.929]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:21:14.443]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:21:15.956]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:17.469]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:18.982]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:19.809]03226C4E
<[0.14]03226C4E
1139626C4E000000

>
>[19:21:19.949]03226C4B
<[0.11]03226C4B
NO DATA

>
>[19:21:20.495]023E01
<[0.11]023E01
NO DATA

>
>[19:21:22.009]023E01
<[0.109]023E01
017EFFFFFFFFFFFF

>
>[19:21:23.522]023E01
<[0.109]023E01
017EFFFFFFFFFFFF

>
>[19:21:25.035]023E01
<[0.094]023E01
017EFFFFFFFFFFFF

>
>[19:21:26.548]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:28.061]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:21:29.575]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:21:31.088]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:32.601]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:34.114]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:35.627]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:21:37.141]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:38.654]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:40.167]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:41.680]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:43.193]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:21:44.707]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:21:46.220]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:47.733]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:49.246]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:21:50.759]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:21:52.273]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:21:53.786]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:21:54.067]022181
<[0.093]022181
1015618156463141

>
>[19:21:54.160]03226C16
<[0.109]03226C16
NO DATA

>
>[19:21:54.269]03226C0A
<[0.125]03226C0A
NO DATA

>
>[19:21:54.394]03226C2F
<[0.109]03226C2F
NO DATA

>
>[19:21:54.503]03226C27
<[0.11]03226C27
NO DATA

>
>[19:21:54.613]03226C09
<[0.109]03226C09
NO DATA

>
>[19:21:54.722]03226C6E
<[0.109]03226C6E
NO DATA

>
>[19:21:54.831]03226C6F
<[0.109]03226C6F
NO DATA

>
>[19:21:54.940]03226C29
<[0.125]03226C29
NO DATA

>
>[19:21:55.065]03226C6D
<[0.109]03226C6D
NO DATA

>
>[19:21:55.174]03226C15
<[0.094]03226C15
04626C1506FFFFFF

>
>[19:21:55.377]023E01
<[0.094]023E01
017EFFFFFFFFFFFF

>
>[19:21:56.812]023E01
<[0.094]023E01
017EFFFFFFFFFFFF

>
>[19:21:58.325]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:21:59.839]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:01.352]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:02.865]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:04.378]023E01
<[0.078]023E01
017EFFFFFFFFFFFF

>
>[19:22:05.891]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:07.405]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:22:08.918]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:10.431]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:11.944]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:13.457]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:14.971]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:22:16.484]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:17.997]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:19.510]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:21.023]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:22.537]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:22:24.050]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:25.563]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:27.076]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:28.589]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:30.103]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:22:31.616]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:33.129]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:34.642]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:36.155]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:37.669]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:39.182]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:40.695]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:42.208]023E01
<[0.078]023E01
017EFFFFFFFFFFFF

>
>[19:22:43.722]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:45.235]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:46.748]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:48.261]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:49.774]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:51.288]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:22:52.801]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:54.314]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:22:55.827]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:22:57.340]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:22:58.854]023E01
<[0.046]023E01
017EFFFFFFFFFFFF

>
>[19:23:00.367]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:23:01.880]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
>[19:23:03.393]023E01
<[0.047]023E01
017EFFFFFFFFFFFF

>
>[19:23:04.906]023E01
<[0.063]023E01
017EFFFFFFFFFFFF

>
>[19:23:06.420]023E01
<[0.062]023E01
017EFFFFFFFFFFFF

>
```

**Comment by cedricp** [comment]
There is communication between the OBD tool and the ECU
- the start diagnostic session is accepted
- Tester present accepted too
- This frame too : 
\< 03-**226C15**
\> 04-**626C1506FFFFFF**

**Comment by premultiply** [comment]
Yes it is but the result in screen is incomplete (no data).
If the same is run on modified ELM327 dongle everything is complete.

**Comment by cedricp** [comment]
Have you tried to connect to another ECU on CAN line 2 ? Maybe a firmware issue on the ELS side

**Comment by premultiply** [comment]
Yes, its the same (bad) result with other known ECUs like Radio.
Unfortunately, I am not able to update the firmware.

**Comment by pakoto** [comment]
Now els27 branch is master?

**Comment by premultiply** [comment]
Yes, looks like Cedric has merged them.

@cedricp I think https://github.com/cedricp/ddt4all/blob/e0cf3c4d63d23ac664196888205a84a6990795a1/elm.py#L692 also needs ST-Handling, isn't it?

**Comment by pakoto** [comment]
Tested with my dacia Dokker and mod els27. Canbus ecus works ok. Medianav ECU dont detect using scan. 

We need yo know how forscan works.

**Comment by pakoto** [comment]
Tested again with last version. Canbus 2 @500k works ok changing options into dacia medianav, but... when i select canbus 2 @500k and push scan ecus, no ecus found. After scan medianav shows NO DATA all options. Need to change canbus 2 500k to other option, change again to canbus 2 500k and push refresh.

**Comment by pakoto** [comment]
[logs.zip](https://github.com/cedricp/ddt4all/files/2158111/logs.zip)

log from session attached. Can 2 500k scan, no ecu found, but changing options works after that. Tested changing boot logo from renault to dacia.

**Comment by premultiply** [comment]
It is on usb_can branch only 4e9eccc013bf47047b5ecc8b2959344ef387f917

**Comment by pakoto** [comment]
yes, tested with this version.

**Comment by pakoto** [comment]
ok, i test with my old "switch" cable and do same: scan don´t detect medianav but change options. Someone with renault gps should test it.

**Comment by pakoto** [comment]
[logs.zip](https://github.com/cedricp/ddt4all/files/2158475/logs.zip)

log (last test)

**Comment by gadget78** [comment]
as this is current, and people are having trouble connecting
(i came here for other reasons, DDCM) ... 

if your trying to communicate with radio/NAV units, the CAN for them is indeed,
pins 12, and 13..
they need to be reversed to the normal wiring (like you get as standard with the gray adapter with ECUSCAN set)
but to put it in simpler terms ... 
you need to put pin 6 to pin 13, and pin 14 to pin 12

hope this helps peoples :)

**Comment by basri06** [comment]
renaulr clio4 aide park kapatma nasıl yapılır

**Comment by pakoto** [comment]
Obdlink SX mod to obtain can2. We can adapt 6-13 and 12-14, but need software support "RTS relay" used in forscan. We need to know how it works to add into ddt4all.

[OBDLINK_SX_FORSCAN_MODS.pdf](https://github.com/cedricp/ddt4all/files/4023250/OBDLINK_SX_FORSCAN_MODS.pdf)

**Comment by premultiply** [comment]
This should be very easy as it just uses the serial RTS (virtual) output pin to switch the CAN lines.

**Comment by pakoto** [comment]
Easy to do mod, but needs changes at software side. 

I don´t know if chinese ELS27 has integrated rts relay switch or different system.

**Comment by kadirayhantr** [comment]
> Bu güncel olduğundan ve insanlar bağlantıda sorun
> yaşıyor (buraya başka nedenlerle geldim, DDCM ve katlanır ayna etkinleştirme) ...
> 
> radyo / NAV üniteleriyle iletişim kurmaya çalışıyorsanız, onlar için CAN gerçekten,
> pim 12 ve 13'tür.
> ancak normal kablolamaya geri döndürülmeleri gerekir (ECUSCAN setli gri adaptörle standart olarak aldığınız gibi)
> ancak daha basit terimlerle ifade
> etmek gerekirse ... pim 6'yı pim 13'e ve pim 14'ü pim 12'ye koymanız gerekir
> 
> Umarım bu insanlara yardımcı olur :)

Hi,

Can you activate power folding mirror on DCM ?   I am trying to do this but I couldn't.  Please guide me if you been success.

Thanks

**Comment by Furtif** [comment]
uses a ubs ok https://github.com/cedricp/ddt4all/issues/834
